### PR TITLE
Add support for aws elasticsearch service

### LINKED ISSUE #39: Add Support for AWS Elasticsearch Service
Pricing page here: https://aws.amazon.com/elasticsearch-service/pricing/

[aws_elasticsearch_domain](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticsearch_domain)

### PR DESCRIPTION
**Pull request by sashman**
Fixes #39 

Adding pricing for 
* Per instance hour
* Storage
  * IOPS
* Dedicated master
* Ultrawarm

image

### CODE REVIEW

File: internal/providers/terraform/aws/elasticsearch_domain.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		iopsVal = decimal.NewFromFloat(ebsOptions.Get("iops").Float())
+
+		if iopsVal.LessThan(decimal.NewFromInt(1)) {
+			iopsVal = decimal.NewFromInt(1)
+		}
+	}
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           "Per instance hour",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/elasticsearch_domain.go:48]: "Instance" should be enough as the unit clarifies it and the calculation already includes the count, so I'm thinking it means "The line item is the instance of your cloud resource, that internally has a few nodes/masters/warm nodes etc"

**Review by alikhajeh1 — COMMENTED** [review]
Very cool! I left 2 comments, I'm working on this table as part of the contributing docs so we can figure-out what names we should use. Will ping you when it's ready so we can discuss - probably tmrw/nxt week.

Screen Shot 2020-09-11 at 2 52 54 PM

File: internal/providers/terraform/aws/elasticsearch_domain.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/infracost/infracost/pkg/schema"
+	"github.com/shopspring/decimal"
+)
+
+func NewElasticsearchDomain(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+
+	region := d.Get("region").String()
+	clusterConfig := d.Get("cluster_config").Array()[0]
+	instanceType := clusterConfig.Get("instance_type").String()
+	instanceCount := clusterConfig.Get("instance_count").Int()
```
[the remark this answers is not in the document]
Comment (sashman) [reply internal/providers/terraform/aws/elasticsearch_domain.go:13]: Yes those are missing, shall I add them as part of this PR @alikhajeh1 ?

File: internal/providers/terraform/aws/elasticsearch_domain.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		iopsVal = decimal.NewFromFloat(ebsOptions.Get("iops").Float())
+
+		if iopsVal.LessThan(decimal.NewFromInt(1)) {
+			iopsVal = decimal.NewFromInt(1)
+		}
+	}
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           "Instance",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/elasticsearch_domain.go:55]: Can we make this Instance (on-demand, instanceType_here)? so it's like the EC2 and Mongo resources. Same thing with other 2 Instance names (master, warm). That should make it clear that this only supports on-demand. We started putting the instanceType in brackets as the EC2 autoscaling group stuff required us to pretty much guess what instanceType we were using for the calculation so we thought we should show it so it's clear what our guess is.

**Comment by alikhajeh1** [comment]
@sashman solid PR - thanks! I'll squash and merge as soon as the tiny issue in my comment is addressed.

File: internal/providers/terraform/aws/elasticsearch_domain.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+		iopsVal = decimal.NewFromFloat(ebsOptions.Get("iops").Float())
+
+		if iopsVal.LessThan(decimal.NewFromInt(1)) {
+			iopsVal = decimal.NewFromInt(1)
+		}
+	}
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Instance on-demand, (%s)", instanceType),
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/elasticsearch_domain.go:57]: I think `fmt.Sprintf("Instance (on-demand, %s)", instanceType),` would make it consistent with EC2/Mongo (same issue with the other two cases)

Comment (sashman) [reply internal/providers/terraform/aws/elasticsearch_domain.go:57]: Haha so close, sorry my bad, should have checked the other resources.

Comment (alikhajeh1) [reply internal/providers/terraform/aws/elasticsearch_domain.go:57]: Thanks, I'll merge as soon as that's done.

**Pull request merged**